src/Files.App/UserControls/StatusCenter/SpeedGraph.cs

```diff
@@ -0,0 +1,185 @@
+			// TODO: unhook
+			this.SizeChanged += OnSizeChanged;
+			this.Unloaded += UserControl_Unloaded;
+			this.ActualThemeChanged += UserControl_ActualThemeChanged;
+		}
+
+		private void OnSizeChanged(object sender, SizeChangedEventArgs e)
+		{
+			if (initialized)
+				return;
+
+			Init();
+
+			// added *after* first load
+			Points.CollectionChanged += PointsChanged;
+			this.Loaded += OnLoaded;
+		}
+
+		private void OnLoaded(object sender, RoutedEventArgs e)
+		{
+			if (Points.Count > 0)
+				UpdateGraph();
+
+			Points.CollectionChanged += PointsChanged;
+		}
+
+		private void UserControl_Unloaded(object sender, RoutedEventArgs e)
+		{
+			Points.CollectionChanged -= PointsChanged;
+		}
+
+		private void PointsChanged(object? sender, NotifyCollectionChangedEventArgs e)
+		{
+			UpdateGraph();
+		}
+
+		private void Init()
+		{
+			compositor = ElementCompositionPreview.GetElementVisual(this).Compositor;
+			rootVisual = compositor.CreateContainerVisual();
+			rootVisual.Size = this.ActualSize;
+			ElementCompositionPreview.SetElementChildVisual(this, rootVisual);
+
+			width = rootVisual.Size.X;
+			height = rootVisual.Size.Y;
+
+			var accentColor = (App.Current.Resources["AccentFillColorDefaultBrush"] as SolidColorBrush)!.Color;
+
+			var backgroundBrush = compositor.CreateColorBrush(accentColor with { A = 0x15 });
+
+			var graphFillBrush = compositor.CreateLinearGradientBrush();
+			graphFillBrush.StartPoint = new(0.5f, 0f);
+			graphFillBrush.EndPoint = new(0.5f, 1f);
+			graphFillBrush.ColorStops.Add(compositor.CreateColorGradientStop(0f, accentColor with { A = 0x70 }));
+			graphFillBrush.ColorStops.Add(compositor.CreateColorGradientStop(1f, accentColor with { A = 0x06 }));
+
+			var graphStrokeBrush = compositor.CreateColorBrush(accentColor);
+
+			var bgVisual = compositor.CreateSpriteVisual();
+			bgVisual.Size = rootVisual.Size;
+			bgVisual.Brush = backgroundBrush;
+
+			var graphVisual = compositor.CreateShapeVisual();
+			graphVisual.Size = rootVisual.Size;
+			graphShape = compositor.CreateSpriteShape();
+			graphShape.FillBrush = graphFillBrush;
+			graphShape.StrokeBrush = graphStrokeBrush;
+			graphShape.StrokeThickness = 1f;
+			graphVisual.Shapes.Add(graphShape);
+
+			var container = compositor.CreateContainerVisual();
+			container.Size = rootVisual.Size;
+			container.Children.InsertAtBottom(bgVisual);
+			container.Children.InsertAtBottom(graphVisual);
+
+			graphClip = compositor.CreateInsetClip();
+			graphClip.RightInset = width;
+			container.Clip = graphClip;
+
+			rootVisual.Children.InsertAtBottom(container);
+
+			line = compositor.CreateSpriteVisual();
+			line.Size = new(width, 1.5f);
+			line.Brush = graphStrokeBrush;
+			rootVisual.Children.InsertAtTop(line);
+
+			highestValue = 0;
+
+			initialized = true;
+		}
+
+		float YValue(float y) => height - (y / highestValue) * (height - 40f) - 4;
+
+		void UpdateGraph()
+		{
+			var geometry = CreatePathFromPoints();
+			graphShape.Geometry = geometry;
```

Comment:
Why is this creating a new geometry instead of changing the `Path` property of the current `CompositionPathGeometry`?

suggestion: maybe a `PathKeyFrameAnimation` can used here to smoothly interpolate between the old path and the newer one?